internal/v1/handler_compose_image.go

```diff
@@ -173,6 +166,116 @@ func (h *Handlers) handleCommonCompose(ctx echo.Context, composeRequest ComposeR
 	}, nil
 }
 
+func buildRepositories(arch *distribution.Architecture, imageType ImageTypes) []composer.Repository {
+	var repositories []composer.Repository
+	for _, r := range arch.Repositories {
+		// If no image type tags are defined for the repo, add the repo
+		contains := len(r.ImageTypeTags) == 0
+		for _, it := range r.ImageTypeTags {
+			if it == string(imageType) {
+				contains = true
+				break
+			}
+		}
+		if contains {
+			repositories = append(repositories, composer.Repository{
+				Baseurl:  r.Baseurl,
+				Metalink: r.Metalink,
+				Rhsm:     common.ToPtr(r.Rhsm),
+				Gpgkey:   r.GpgKey,
+				CheckGpg: r.CheckGpg,
+			})
+		}
+	}
+	return repositories
+}
+
+func (h *Handlers) buildRepositorySnapshots(ctx echo.Context, arch *distribution.Architecture, imageType ImageTypes, snapshotDate string) ([]composer.Repository, error) {
+	date, err := time.Parse(time.RFC3339, snapshotDate)
+	if err != nil {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, fmt.Sprintf("Snapshot date %s is not in RFC3339 format", snapshotDate))
+	}
+
+	repoURLs := []string{}
+	for _, r := range arch.Repositories {
+		contains := len(r.ImageTypeTags) == 0
+		for _, it := range r.ImageTypeTags {
+			if it == string(imageType) {
+				contains = true
+				break
+			}
+		}
+		if contains {
+			repoURLs = append(repoURLs, *r.Baseurl)
+		}
+	}
+
+	repoUUIDs := []string{}
+	repoMap := map[string]content_sources.ApiRepositoryResponse{}
+	resp, err := h.server.csClient.GetRepositories(ctx.Request().Context(), repoURLs, false)
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(ctx, resp.Body)
+
+	var csRepos content_sources.ApiRepositoryCollectionResponse
+	err = json.NewDecoder(resp.Body).Decode(&csRepos)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, repo := range *csRepos.Data {
+		repoUUIDs = append(repoUUIDs, *repo.Uuid)
+		repoMap[*repo.Uuid] = repo
+	}
+
+	snapResp, err := h.server.csClient.GetSnapshotsForDate(ctx.Request().Context(), content_sources.ApiListSnapshotByDateRequest{
+		Date:            common.ToPtr(date.UTC().Format(time.DateOnly)),
+		RepositoryUuids: common.ToPtr(repoUUIDs),
+	})
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(ctx, snapResp.Body)
+
+	var csSnapshots []content_sources.ApiListSnapshotByDateResponse
```

Comment:
Ugh, it's a shame that we have root array here, that smells, should we ask Content to change this before we start depending on it?

Reply:
maybe, not sure. Could also merge and ask at the same time :innocent:

Reply:
It's updated, can we use the new API :pray: :)